packages/react-urql/src/hooks/useQuery.ts

```diff
@@ -41,70 +40,62 @@ export type UseQueryResponse<Data = any, Variables = object> = [
   (opts?: Partial<OperationContext>) => void
 ];
 
+const notFetching = initialState;
+const fetching = { ...initialState, fetching: true };
+
 const isSuspense = (client: Client, context?: Partial<OperationContext>) =>
   client.suspense && (!context || context.suspense !== false);
 
-let currentInit = false;
-
 export function useQuery<Data = any, Variables = object>(
   args: UseQueryArgs<Variables, Data>
 ): UseQueryResponse<Data, Variables> {
   const client = useClient();
-  const cache = getCacheForClient(client);
-  const suspense = isSuspense(client, args.context);
   const request = useRequest<Data, Variables>(args.query, args.variables);
+  const suspense = isSuspense(client, args.context);
```

Comment:
We can potentially lift a limitation here for suspense. Previously it wasn't possible to disable `suspense` using `executeQuery`; however, we can circumvent this by deriving this in `getSnapshot` and keeping an extra suspense flag around in `executeQuery` + `meta`